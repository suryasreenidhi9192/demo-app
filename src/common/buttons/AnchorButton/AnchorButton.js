import React from 'react'
import PropTypes from 'prop-types';

import { curryWithPreventDefault } from '../../../common/shared/utils/events'

import './AnchorButton.scss'

/**
 * @public
 * A component to create HTML anchors with shorcuts for underlining and prevent default on click handlers
 * @param  {Node|String}    props.children  React Component or string to render in the button/link
 * @param  {Boolean}        props.disabled  Disables the button/link
 * @param  {Function}       props.onClick   Fires curryWithPreventDefault util wrapping this callback,
                                            allowing `href` to be set for crawlers or screen readers
 * @param  {Boolean}        props.underline Sets the link to underline or not
 * @param  {...Array}       props.rest      Rest params to pass onto html anchor
 * @return {Node}                           React Component
 */
const AnchorButton = ({ children, disabled, onClick, underline, ...rest }) => {

    const extraProps = {}
    if (!!disabled) extraProps.disabled = 'disabled'
    const disabledClass = (!!disabled) ? 'disabled' : ''
    const underlineClass = underline ? 'underline' : ''
    const cls = `anchor-button ${disabledClass} ${underlineClass}`

    return (

        <a
            href="#"
            role="button"
            tabIndex="0"
            className={ cls }
            onClick={ curryWithPreventDefault(onClick) }
            {...extraProps}
            {...rest}
        >
            { children }
        </a>

    )

}

AnchorButton.propTypes = {
    /** React Component or string to render in the button/link */
    children: PropTypes.node,
    /** Disables the button/link */
    disabled: PropTypes.bool,
    /** Sets the link to underline or not */
    underline: PropTypes.bool,
    /** fires curryWithPreventDefault util wrapping this callback,
    allowing `href` to be set for crawlers or screen readers */
    onClick: PropTypes.func
};

export default AnchorButton
