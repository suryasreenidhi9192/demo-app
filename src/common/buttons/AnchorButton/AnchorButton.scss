@import "../../shared/css/variables";

.anchor-button {

    font-family: $reg;
    color: var(--action);
    cursor: pointer;
    user-select: none;

    &:focus, &:hover {
        text-decoration: underline;
        color: var(--action-hover);
    }

    &:active {
        color: var(--action-active);
    }

    &.underline {
        text-decoration: underline;
    }

    &.disabled {
        opacity: 0.5;
        text-decoration: none;
        pointer-events: none;
     }

}

