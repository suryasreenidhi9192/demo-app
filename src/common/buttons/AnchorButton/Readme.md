### Simple AnchorButton
```jsx
    <AnchorButton> Click Here</AnchorButton>
```

### Underlined AnchorButton
```jsx
    <AnchorButton underline> Click Here</AnchorButton>
```

### Disabled AnchorButton
```jsx
    <AnchorButton disabled> Click Here</AnchorButton>
```
