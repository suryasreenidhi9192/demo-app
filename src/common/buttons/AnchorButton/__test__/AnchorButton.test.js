import React from 'react'
import { mount } from 'enzyme';
import { matchSnapshot, testComponentOnClick } from '../../../../../__test__/__utils__'

import AnchorButton from '../'

const props = {
    onClick: jest.fn(),
    children: (<span>Click Here, Ron ;)</span>)
}

const underlineProps = {
    ...props,
    underline: true
}

const disabledProps = {
    ...props,
    disabled: true
}

const testClickEventMethodTruthiness = eventMethodName => () => {
    const click = (event) => {
        expect(event[eventMethodName]()).toBe(true)
    }
    const clickTestLinkData = { onClick: click, ...props }
    const clickTestNavItem = mount(
        <AnchorButton { ...clickTestLinkData } />
    )
    clickTestNavItem.simulate('click')
}

describe('AnchorButton:.......................SnapShot Testing:', () => {
    test(
        'Default AnchorButton Matches jest snapshot',
        matchSnapshot(AnchorButton, props)
    )
    test(
        'Underlined AnchorButton Matches jest snapshot',
        matchSnapshot(AnchorButton, underlineProps)
    )
    test(
        'Disabled AnchorButton Matches jest snapshot',
        matchSnapshot(AnchorButton, disabledProps)
    )
})

describe('AnchorButton:.......................Event Handling:', () => {
    test(
        'Clicking AnchorButton fires the handler',
        testComponentOnClick(AnchorButton, props)
    )
    test(
        'Clicking AnchorButton fires the handler with default prevented',
        testClickEventMethodTruthiness('isDefaultPrevented')
    )

})
