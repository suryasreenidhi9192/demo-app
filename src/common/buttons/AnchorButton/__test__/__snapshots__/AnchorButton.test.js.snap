// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnchorButton:.......................SnapShot Testing: Default AnchorButton Matches jest snapshot 1`] = `
<a
  className="anchor-button  "
  href="#"
  onClick={[Function]}
  role="button"
  tabIndex="0"
>
  <span>
    Click Here, Ron ;)
  </span>
</a>
`;

exports[`AnchorButton:.......................SnapShot Testing: Disabled AnchorButton Matches jest snapshot 1`] = `
<a
  className="anchor-button disabled "
  disabled="disabled"
  href="#"
  onClick={[Function]}
  role="button"
  tabIndex="0"
>
  <span>
    Click Here, Ron ;)
  </span>
</a>
`;

exports[`AnchorButton:.......................SnapShot Testing: Underlined AnchorButton Matches jest snapshot 1`] = `
<a
  className="anchor-button  underline"
  href="#"
  onClick={[Function]}
  role="button"
  tabIndex="0"
>
  <span>
    Click Here, Ron ;)
  </span>
</a>
`;
